<!--
 Copyright 2020 Carnegie Mellon University

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-->

<launch>
  <arg name="output" default="log"/>
  <arg name="db_path" />
  <arg name="init_speed" default=""/>
  <arg name="config_user" value="cabot"/>
  <arg name="anchor_file" default=""/>
  <arg name="language" default="en"/>

  <include file="$(find mongodb_store)/launch/mongodb_store.launch">
    <arg name="db_path" value="$(arg db_path)"/>
  </include>

  <group ns="cabot">

    <node pkg="cabot_ui" type="tts_node.py" name="tts_node" output="$(arg output)">
      <rosparam file="$(find cabot_ui)/key/key.yaml" command="load"/>
    </node>
    <!--
	<node pkg="cabot_ui" type="cabot_ble.py" name="cabot_ble_node" output="$(arg output)" ns="cabot">
	</node>
    <node pkg="cabot_ui" type="cabot_menu_node.py"
    -->

    <node pkg="cabot_ui" type="cabot_ui_manager.py" name="cabot_ui_manager" output="$(arg output)">
      <param name="config_user" value="$(arg config_user)"/>
      <param name="init_speed" value="$(arg init_speed)"/>
      <param name="anchor_file" value="$(arg anchor_file)" />
      <param name="language" value="$(arg language)" />
      <rosparam file="$(find cabot_ui)/menu/menu.yaml" command="load"/>
    </node>

  </group>

  <!-- integrated into cabot_ui
  <node pkg="cabot_ui" type="navcognode.py" name="cabot_ui" output="$(arg output)">
    <param name="anchor_file" value="$(arg anchor_file)" />
    </node>
    -->

</launch>
